include/onnxruntime/core/graph/node_arg.h

```diff
@@ -68,13 +68,13 @@ class NodeArg {
   @param strict If true, the shape update will fail if there are incompatible values. 
                 If false, will be lenient and merge only shape info that can be validly processed.
   @returns Success unless there is existing type or shape info that can't be successfully updated. */
-  common::Status UpdateTypeAndShape(const ONNX_NAMESPACE::TypeProto& input_type, bool strict = true);
+  common::Status UpdateTypeAndShape(const ONNX_NAMESPACE::TypeProto& input_type, bool strict, const logging::Logger* logger);
```

Comment:
>const logging::Logger* logger [](start = 94, length = 29)

Shouldn't we enforce that there is always a logger available? Much simpler than having 'if (logger != nullptr)' checks everywhere.

Reply:
Thanks. But that requires a big change.  I'm doing it.